javaparser-core/src/main/java/com/github/javaparser/printer/lexicalpreservation/LexicalPreservingPrinter.java

```diff
@@ -164,27 +170,43 @@ public void concretePropertyChange(Node observedNode, ObservableProperty propert
             LEXICAL_DIFFERENCE_CALCULATOR.calculatePropertyChange(nodeText, observedNode, property, oldValue, newValue);
         }
 
-        private List<TokenTextElement> getMatchingTokenTextElements(JavadocComment oldValue, NodeText nodeText) {
-            JavadocComment javadocComment = oldValue;
-            List<TokenTextElement> matchingTokens = nodeText.getElements().stream()
+        private List<TokenTextElement> getMatchingTokenTextElements(Comment oldValue, NodeText nodeText) {
+            List<TokenTextElement> matchingTokens;
+            
+            if (oldValue instanceof JavadocComment) {
+                matchingTokens = nodeText.getElements().stream()
                     .filter(e -> e.isToken(JAVADOC_COMMENT))
                     .map(e -> (TokenTextElement) e)
-                    .filter(t -> t.getText().equals("/**" + javadocComment.getContent() + "*/"))
+                    .filter(t -> t.getText().equals("/**" + oldValue.getContent() + "*/"))
```

Comment:
Do we have a utility method to produce this string?

Reply:
I don't think there is a utility method for this. I saw this way of assembling the string used in other parts of the code (like the prettyPrintingTextNode method) so I simply did the same thing here.